backend/fms_core/viewsets/container.py

```diff
@@ -185,9 +186,11 @@ def search(self, _request):
                 query.add(Q(name__icontains=search_input), Q.OR)
                 query.add(Q(id__icontains=search_input), Q.OR)
             if is_parent:
-                query.add(Q(kind__in=PARENT_CONTAINER_KINDS), Q.AND)
+                kinds = [kind for kind in PARENT_CONTAINER_KINDS if kind not in except_kinds]
+                query.add(Q(kind__in=kinds), Q.AND)
             if is_sample_holding:
-                query.add(Q(kind__in=SAMPLE_CONTAINER_KINDS), Q.AND)
+                kinds = [kind for kind in SAMPLE_CONTAINER_KINDS if kind not in except_kinds]
```

Comment:
maybe `kinds` can be assigned outside the `if`s?

Reply:
Mmm  i don't see much gain you would still need to use an `if` on the KINDS to choose from ... and the query only has kind for those 2 cases ...

Reply:
Oh woops didn't notice they use different constants